### PR TITLE
Makefile changes for Windows and easier development

### PR DESCRIPTION
**Pull request by jolheiser**
This just adds your `go/bin` to PATH and changes `generate-stylesheets` to use `npx` which is bundled with npm version 5.2+  
If npx isn't found, it asks the user to update their npm to version 5.2+  
  
This makes it easier for Windows devs, since Windows didn't like `node_modules/.bin/lessc` etc complaining about the path separator.   
As well, it covers if a user hasn't already added their `go/bin` to their PATH, which would then make things like `lint` and `misspell-check` not work.

### CODE REVIEW

**Comment by jolheiser** [comment]
@silverwind Any suggestions or things you don't like/would change?

**Comment by silverwind** [comment]
If you remove `node_modules/.bin` it will use the global binaries instead of the ones we depend on. I suggest instead do `npx lessc` and similar. It will require node.js >= 8, but seeing as 6 is going EOL in 2 months, it should be acceptable.

**Comment by jolheiser** [comment]
Ah, okay. I had first tried that, but npx didn't seem to play nice with me. Any tips?

To clarify, `npx lessc` didn't seem to work, it wanted `npx less`.
And I never did get postcss to work for me.

**Comment by silverwind** [comment]
Never tried it on Windows, but I assume it should just work. Tried updating node and/or npm?

**Comment by jolheiser** [comment]
Using the current LTS of Node. Did a full reinstall for testing.
I'll mess with it a little more tonight and see what I can come up with.

On a separate note, since the local node modules is first in the path wouldn't it use that?

**Comment by silverwind** [comment]
Would probably work, not sure if `$PATH` can really work properly with relative paths. npx uses node_modules and falls back to downloading the module if not found.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 DIST := dist
 IMPORT := code.gitea.io/gitea
 
+export PATH := $(HOME)/go/bin:node_modules/.bin:$(PATH)
```
Comment (silverwind) [inline Makefile]: I would at least do`$GOPATH/bin`. Assuming it's `$HOME/go/bin` is bound to fail, especially on servers.

Comment (zeripath) [reply Makefile]: Actually `$(go env GOPATH)/bin` might be even better.

**Comment by jolheiser** [comment]
@silverwind Alright, it's using npx now. Turns out my npm path was messed up for some reason and that messed with npx  
  
@zeripath I also changed to using `$(go env GOPATH)` to add to the path

@techknowlogick Pinging you as well since you had brought it up in the Discord. 👍

**Review by silverwind — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,8 @@
 DIST := dist
 IMPORT := code.gitea.io/gitea
 
+export PATH := $(go env GOPATH)/bin:$(PATH)
```
Comment (techknowlogick) [inline Makefile]: A very tiny nit, could you use `$(GO)` in lieu of `go`. You may need to move this declaration down a few lines below where `GO` is defined.

Comment (jolheiser) [reply Makefile]: Done

**Review by techknowlogick — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Oh before we merge this @jolheiser could you check the hacking documentation to make sure it doesn't conflict with this?

I think it's fine and if anything we suggest doing stuff that this makes unnecessary but just to be sure...

**Comment by jolheiser** [comment]
The hacking documentation recommends adding Go to your path, which is unnecessary with this change, however as you said that's not necessarily a bad thing. This would just double it up.  
  
As for the npx changes, currently the hacking documentation says to run `npm install` but doesn't actually mention that the user will need node (which would also contain npm).  
The changes would check for npx and alert the user to update to a compatible version of npm, though.  
  
Should I add a comment about installing node/npm before working with the stylesheets?

**Comment by silverwind** [comment]
> Should I add a comment about installing node/npm before working with the stylesheets?

Yes, I'd recommend we require Node.js >= 8.0 with npm included (to my knowledge, all distributions of Node.js include npm).

**Comment by jolheiser** [comment]
Done

**Review by zeripath — APPROVED** [review]

**Comment by silverwind** [comment]
Would you mind also adding the version [here](https://github.com/go-gitea/gitea/blob/master/CONTRIBUTING.md#building-gitea)?

**Comment by jolheiser** [comment]
Done

**Pull request merged**